h2o-py/tests/testdir_algos/automl/pyunit_automl_args.py

```diff
@@ -409,14 +409,21 @@ def test_max_runtime_secs_can_be_set_in_combination_with_max_models_and_max_mode
 
 def test_max_runtime_secs_can_be_set_in_combination_with_max_models_and_max_runtime_wins():
     ds = import_dataset()
-    aml = H2OAutoML(project_name="py_all_stopping_constraints", seed=1, max_models=20, max_runtime_secs=12)
+    aml_probe = H2OAutoML(project_name="py_all_stopping_constraints_probe",
+                          max_models=5,
+                          seed=1, exclude_algos=["XGBoost", "StackedEnsemble"])
+    aml_probe.train(y=ds['target'], training_frame=ds['train'])
+    probe_duration = int(aml_probe.training_info['duration_secs'])
+
+    aml = H2OAutoML(project_name="py_all_stopping_constraints",
+                    max_models=20, max_runtime_secs=probe_duration,
+                    seed=1, exclude_algos=["XGBoost", "StackedEnsemble"])
     aml.train(y=ds['target'], training_frame=ds['train'])
-    max_runtime = aml._build_resp['build_control']['stopping_criteria']['max_runtime_secs']
-    max_models = aml._build_resp['build_control']['stopping_criteria']['max_models']
-    assert max_runtime == 12
-    assert max_models == 20
+
+    assert aml._build_resp['build_control']['stopping_criteria']['max_runtime_secs'] == probe_duration
+    assert aml._build_resp['build_control']['stopping_criteria']['max_models'] == 20
     assert aml.leaderboard.nrows < 20
-    assert int(aml.training_info['duration_secs']) < 2*max_runtime  # being generous to avoid errors on slow Jenkins
+    assert int(aml.training_info['duration_secs']) < 2*probe_duration  # being generous to avoid errors on slow Jenkins
```

Comment:
the assertion on `nrows` was wrong by contract, as there's nothing preventing us to have 20 models on a very fast machine, it should be:
```
    assert aml.leaderboard.nrows <= 20
```

As we have now a probe, maybe we can be less generous.
This would be my suggestion:
aml_probe: max_models=5   ->  probe_duration
aml_tested: max_models=5, max_runtime_secs=probe_duration/2  -> tested_duration
assert aml_tested.leaderboard.nrows <= 5
assert int(aml_tested.training_info['duration_secs']) < probe_duration

As we use the same `max_models` in both, the 2 assertions should be valid by contract.